adafruit_pybadger.py

```diff
@@ -0,0 +1,415 @@
+        self._speaker_enable.switch_to_output(value=False)
+        self._sample = None
+        self._sine_wave = None
+        self._sine_wave_sample = None
+
+    def _check_for_movement(self, movement_threshold=10):
+        """Checks to see if board is moving. Used to auto-dim display when not moving."""
+        current_accelerometer = self.acceleration
+        if self._last_accelerometer is None:
+            self._last_accelerometer = current_accelerometer
+            return False
+        acceleration_delta = sum([abs(self._last_accelerometer[n] - current_accelerometer[n]) for n
+                                  in range(3)])
+        self._last_accelerometer = current_accelerometer
+        return acceleration_delta > movement_threshold
+
+    def auto_dim_display(self, delay=5.0, movement_threshold=10):
+        """Auto-dim the display when board is not moving.
+
+        :param int delay: Time in seconds before display auto-dims after movement has ceased.
+        :param int movement_threshold: Threshold required for movement to be considered stopped.
+                                       Change to increase or decrease sensitivity.
+
+        """
+        if not self._check_for_movement(movement_threshold=movement_threshold):
+            current_time = time.monotonic()
+            if current_time - self._start_time > delay:
+                self.display.brightness = 0.1
+                self._start_time = current_time
+        else:
+            self.display.brightness = 1
+
+    @property
+    def pixels(self):
+        """Sequence like object representing the NeoPixels on the board."""
+        return self._neopixels
+
+    @property
+    def joystick(self):
+        """The joystick on the PyGamer."""
+        if hasattr(board, "JOYSTICK_X"):
+            x = self._pygamer_joystick_x.value
+            y = self._pygamer_joystick_y.value
+            return x, y
+        raise RuntimeError("This board does not have a built in joystick.")
+
+    @property
+    def button(self):
+        """The buttons on the board.
+
+        Example use:
+
+        .. code-block:: python
+
+        from adafruit_pybadger import PyBadger
+
+        pybadger = PyBadger()
+
+        while True:
+            if pybadger.button.a:
+                print("Button A")
+            elif pybadger.button.b:
+                print("Button B")
+            elif pybadger.button.start:
+                print("Button start")
+            elif pybadger.button.select:
+                print("Button select")
+
+        """
+        button_values = self._buttons.get_pressed()
+        return Buttons(*[button_values & button for button in
+                         (PyBadger.BUTTON_B, PyBadger.BUTTON_A, PyBadger.BUTTON_START,
+                          PyBadger.BUTTON_SELECT, PyBadger.BUTTON_RIGHT,
+                          PyBadger.BUTTON_DOWN, PyBadger.BUTTON_UP, PyBadger.BUTTON_LEFT)])
```

Comment:
Should this handle the joystick as a directional on pygamer as well?

Why have one property for all buttons instead of 8 separate ones to match circuitplayground?

Reply:
I'm not sure how to make it handle the joystick. There's a separate property for the joystick.

I initially had it in separate properties, however calling `get_pressed` multiple times clears the previous button press. This seemed more efficient and allows for multiple button presses to be maintained.

Reply:
I wonder if the multiple press tracking isn't needed for this simple library. Your example code calls get_pressed multiple times anyway.

I imagine having the joystick count as a directional button would be more beneficial because the client code will work on PyGamer and PyBadge without modification.